bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java

```diff
@@ -116,9 +121,14 @@ public void initialize() {
         }
 
         logger.debug("Creating BlueZ adapter with address '{}'", address);
-        for (tinyb.BluetoothAdapter a : BluetoothManager.getBluetoothManager().getAdapters()) {
-            if (a.getAddress().equals(address.toString())) {
-                adapter = a;
+
+        for (tinyb.BluetoothAdapter adapter : manager.getAdapters()) {
+            if (adapter == null) {
+                logger.warn("got null adapter from bluetooth manager");
```

Comment:
In which case would you get a 'null' adapter from the bluetooth manager? This could help someone using this binding to debug any problems.

Reply:
Line 168: of [BluetoothManager.cpp](https://github.com/intel-iot-devkit/tinyb/blob/master/src/BluetoothManager.cpp). How could we ever get a null pointer from the adapter list? Could it be a bug in tinyb?

Reply:
We don't have null annotations available for tinyb. IMO we should not rely on implementation details.

Reply:
I don't agree, because you'd also have to check for null in `adapter.getAddress()` and anything else you retrieve from the native code. I'd check the problem origin of these null adapters, because it signals a larger underlying problem.